grant-project/grant-monorepo
# Web3 Only Auth + MetaMask future proofing

Closes #180. Closes #185.

### What This Does

Narrows us back down to web3-only site interaction, so no more hardware wallets or custom key signing. Nothing was removed, just commented out for now.

This also gets us ready for MetaMask's `ethereum.enable()` requirement for viewing accounts. Works in a backwards-compatible way that doesn't run it for web3 providers that already provide accounts.

### Steps to Test

* Ensure auth works with existing metamask version
* Ensure auth errors correctly for non-web3 browsers
* Install the latest [dev metamask from here](https://github.com/MetaMask/metamask-extension/pull/4703#issuecomment-436008161) and go through auth
    * Cancel the auth, confirm an error correctly shows
    * Accept the auth request, confirm auth continues as normal
    * Log out and refresh the page, confirm you didn't have to auth metamask again for it to show your account 

## Screenshots

### Auth popup
<img width="472" alt="screen shot 2018-11-05 at 4 40 21 pm" src="https://user-images.githubusercontent.com/649992/48029270-d1801980-e11b-11e8-8f2c-1f07ea3cf38b.png">

### While the auth popup is up, show a loader
<img width="625" alt="screen shot 2018-11-05 at 4 44 12 pm" src="https://user-images.githubusercontent.com/649992/48029273-d5ac3700-e11b-11e8-9aa1-74327105c738.png">

### If you cancel it, you get an error. But you can try again.
<img width="621" alt="screen shot 2018-11-05 at 4 44 39 pm" src="https://user-images.githubusercontent.com/649992/48029278-d775fa80-e11b-11e8-8440-6a1cc385fbaf.png">
